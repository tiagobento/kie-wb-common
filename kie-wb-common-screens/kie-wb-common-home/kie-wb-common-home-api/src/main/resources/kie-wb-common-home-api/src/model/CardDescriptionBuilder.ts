import {
  CardDescription,
  CardDescriptionElement,
  CardDescriptionLinkElement,
  CardDescriptionTextElement
} from "./CardDescription";

export class CardDescriptionBuilder {
  private static readonly linkTokenRegex = new RegExp(/{(\d+)}/g);

  private readonly textMask: string;
  private readonly links: CardDescriptionLinkElement[];

  constructor(textMask: string) {
    this.textMask = textMask;
    this.links = [];
  }

  public addLinkIf(predicate: () => boolean, text: string, targetId: string): CardDescriptionBuilder {
    if (!predicate()) {
      return this;
    }
    return this.addLink(text, targetId);
  }

  public addLink(text: string, targetId: string): CardDescriptionBuilder {
    this.links.push(new CardDescriptionLinkElement(text, targetId));
    return this;
  }

  public build(): CardDescription {
    const linkTokens = this.textMask.match(CardDescriptionBuilder.linkTokenRegex);

    if (linkTokens === null || linkTokens.length === 0) {
      return new CardDescription([new CardDescriptionTextElement(this.textMask)]);
    }

    const elements: CardDescriptionElement[] = [];
    for (let i = 0; i < linkTokens.length; i++) {
      const textBeforeLink = this.getTextUntilLinkAt(this.textMask, linkTokens, i);

      if (textBeforeLink.length > 0) {
        elements.push(new CardDescriptionTextElement(textBeforeLink));
      }

      elements.push(this.links[this.linkIdxFromToken(linkTokens[i])]);
    }

    const lastElement = new CardDescriptionTextElement(
      this.getTextUntilLinkAt(this.textMask, linkTokens, linkTokens.length)
    );

    if (lastElement.text.length > 0) {
      elements.push(lastElement);
    }

    return new CardDescription(elements);
  }

  private getTextUntilLinkAt(textMask: string, tokens: RegExpMatchArray, i: number) {
    const prevLinkIdx = i === 0 ? 0 : this.linkIdxFromToken(tokens[i - 1]);
    const prevLinkTkn = `{${prevLinkIdx}}`;

    const textTokenStart = i === 0 ? 0 : textMask.indexOf(prevLinkTkn) + prevLinkTkn.length;
    const textTokenEnd = i === tokens.length ? textMask.length : textMask.indexOf(tokens[i]);

    return textMask.substring(textTokenStart, textTokenEnd);
  }

  private linkIdxFromToken(tkn: string) {
    return parseInt(tkn.replace("{", "").replace("}", ""), 10);
  }
}
