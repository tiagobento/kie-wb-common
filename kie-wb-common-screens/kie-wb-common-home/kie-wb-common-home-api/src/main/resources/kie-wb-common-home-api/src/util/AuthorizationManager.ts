export class AuthorizationManager {
  public static hasAccessToPerspective(perspectiveId: string): boolean {
    return (window as any).AppFormer.HomeModelAuthorizationManager.authorize(perspectiveId);
  }
}
