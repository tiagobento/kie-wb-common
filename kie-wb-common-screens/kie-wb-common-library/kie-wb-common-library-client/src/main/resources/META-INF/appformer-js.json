{
  "name":"kie-wb-common-library-client",
  "type": "lib",
  "main":"./dist/index.js",
  "decorators": {},
  "components": ["LibraryOrganizationalUnitsScreen"]
}