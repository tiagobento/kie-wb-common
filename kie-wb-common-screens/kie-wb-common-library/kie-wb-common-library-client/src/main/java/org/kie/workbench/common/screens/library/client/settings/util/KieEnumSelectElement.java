/*
 * Copyright (C) 2017 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.kie.workbench.common.screens.library.client.settings.util;

import java.util.Arrays;
import java.util.List;
import java.util.function.Consumer;

import javax.enterprise.context.Dependent;
import javax.inject.Inject;

import elemental2.dom.Element;
import org.jboss.errai.ui.client.local.spi.TranslationService;

import static java.util.stream.Collectors.toList;

@Dependent
public class KieEnumSelectElement<T extends Enum<T>> {

    private final KieSelectElement kieSelectElement;
    private final TranslationService translationService;

    Class<T> componentType;

    @Inject
    public KieEnumSelectElement(final KieSelectElement kieSelectElement,
                                final TranslationService translationService) {

        this.kieSelectElement = kieSelectElement;
        this.translationService = translationService;
    }

    @SuppressWarnings("unchecked")
    public void setup(final Element element,
                      final T[] values,
                      final T initialValue,
                      final Consumer<T> onChange) {

        componentType = (Class<T>) values.getClass().getComponentType();

        kieSelectElement.setup(
                element,
                buildOptions(values),
                initialValue.name(),
                name -> onChange.accept(toEnum(name)));
    }

    List<KieSelectElement.Option> buildOptions(final T[] values) {
        return Arrays.stream(values).map(this::newOption).collect(toList());
    }

    KieSelectElement.Option newOption(final T e) {
        return new KieSelectElement.Option(getLabel(e), e.name());
    }

    String getLabel(final T e) {
        return translationService.format(e.name());
    }

    public T getValue() {
        return toEnum(kieSelectElement.getValue());
    }

    T toEnum(final String value) {
        return Enum.valueOf(componentType, value);
    }
}
