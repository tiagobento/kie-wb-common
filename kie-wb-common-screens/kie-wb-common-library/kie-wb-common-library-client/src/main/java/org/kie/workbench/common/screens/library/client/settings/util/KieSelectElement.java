/*
 * Copyright (C) 2017 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.kie.workbench.common.screens.library.client.settings.util;

import java.util.List;
import java.util.function.Consumer;

import javax.annotation.PostConstruct;
import javax.enterprise.context.Dependent;
import javax.inject.Inject;

import elemental2.dom.Element;
import elemental2.dom.HTMLSelectElement;
import org.jboss.errai.ioc.client.api.ManagedInstance;
import org.uberfire.client.mvp.UberElemental;

@Dependent
public class KieSelectElement {

    private final View view;
    private final OptionsListPresenter optionsListPresenter;

    Consumer<String> onChange;

    @Inject
    public KieSelectElement(final View view,
                            final OptionsListPresenter optionsListPresenter) {
        this.view = view;
        this.optionsListPresenter = optionsListPresenter;
        this.onChange = i -> {
        };
    }

    @PostConstruct
    public void init() {
        view.init(this);
    }

    public void setup(final Element element,
                      final List<Option> options,
                      final String initialValue,
                      final Consumer<String> onChange) {

        element.innerHTML = "";
        element.appendChild(view.getElement());

        optionsListPresenter.setup(
                view.getSelect(),
                options,
                (item, presenter) -> presenter.setup(item, this));

        view.setValue(initialValue);

        this.onChange = onChange;

        view.initSelect();
    }

    public void onChange() {
        this.onChange.accept(getValue());
    }

    public void setValue(final String value) {
        view.setValue(value);
    }

    public String getValue() {
        return view.getValue();
    }

    public interface View extends UberElemental<KieSelectElement> {

        HTMLSelectElement getSelect();

        void initSelect();

        void setValue(final String value);

        String getValue();
    }

    public static class Option {

        public final String value;
        public final String label;

        public Option(final String label, final String value) {
            this.label = label;
            this.value = value;
        }
    }

    @Dependent
    public static class OptionElement extends ListItemPresenter<Option, KieSelectElement, KieSelectElementView.Option> {

        private Option option;

        @Inject
        public OptionElement(final KieSelectElementView.Option view) {
            super(view);
        }

        @Override
        public OptionElement setup(final Option option,
                                   final KieSelectElement parentPresenter) {
            this.option = option;

            view.init(this);
            view.setLabel(option.label);
            view.setValue(option.value);

            return this;
        }

        @Override
        public Option getObject() {
            return option;
        }
    }

    @Dependent
    public static class OptionsListPresenter extends ListPresenter<Option, OptionElement> {

        @Inject
        public OptionsListPresenter(final ManagedInstance<OptionElement> itemPresenters) {
            super(itemPresenters);
        }
    }
}
