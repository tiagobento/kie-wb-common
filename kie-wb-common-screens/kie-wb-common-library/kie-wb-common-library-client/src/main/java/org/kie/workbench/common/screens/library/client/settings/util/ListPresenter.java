/*
 * Copyright (C) 2017 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.kie.workbench.common.screens.library.client.settings.util;

import java.util.ArrayList;
import java.util.List;
import java.util.function.BiConsumer;

import elemental2.dom.Element;
import org.jboss.errai.ioc.client.api.ManagedInstance;

import static java.util.stream.Collectors.toList;

public abstract class ListPresenter<T, P extends ListItemPresenter<T, ?, ?>> {

    private final ManagedInstance<P> itemPresenters;

    private List<T> objects;
    private List<P> presenters;
    private Element listElement;
    private BiConsumer<T, P> itemPresenterConfigurator;

    public ListPresenter(final ManagedInstance<P> itemPresenters) {
        this.itemPresenters = itemPresenters;
    }

    public void setup(final Element listElement,
                      final List<T> objects,
                      final BiConsumer<T, P> itemPresenterConfigurator) {

        this.objects = objects;
        this.presenters = new ArrayList<>();
        this.listElement = listElement;
        this.itemPresenterConfigurator = itemPresenterConfigurator;

        this.listElement.innerHTML = "";
        this.objects.forEach(this::addToListElement);
    }

    public void setupWithPresenters(final Element listElement,
                                    final List<P> presenters,
                                    final BiConsumer<T, P> itemPresenterConfigurator) {

        this.objects = presenters.stream().map(p -> p.getObject()).collect(toList());
        this.presenters = presenters;
        this.listElement = listElement;
        this.itemPresenterConfigurator = itemPresenterConfigurator;

        this.listElement.innerHTML = "";
        presenters.forEach(this::addPresenter);
    }

    public void add(final T o) {
        addToListElement(o);
        objects.add(o);
    }

    protected void addToListElement(final T o) {
        addPresenter(newPresenterFor(o));
    }

    protected void addPresenter(final P presenter) {
        presenters.add(presenter);
        listElement.appendChild(presenter.getView().getElement());
    }

    protected P newPresenterFor(final T o) {
        final P listItemPresenter = this.itemPresenters.get();
        listItemPresenter.setListPresenter(this);
        itemPresenterConfigurator.accept(o, listItemPresenter);
        return listItemPresenter;
    }

    public void remove(final ListItemPresenter<T, ?, ?> listItemPresenter) {
        objects.remove(listItemPresenter.getObject());
        listElement.removeChild(listItemPresenter.getView().getElement());
    }

    public List<T> getObjectsList() {
        return objects;
    }

    public List<P> getPresenters() {
        return presenters;
    }

    @Override
    public int hashCode() {
        return objects.hashCode();
    }
}
