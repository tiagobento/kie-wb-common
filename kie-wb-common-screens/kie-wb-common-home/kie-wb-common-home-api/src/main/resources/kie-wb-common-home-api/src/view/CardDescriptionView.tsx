import * as React from "react";
import * as AppFormer from "appformer-js";
import { CardDescription, CardDescriptionLinkElement, CardDescriptionTextElement } from "../model";
import { AuthorizationManager } from "../util";

interface Props {
  description: CardDescription;
}

export class CardDescriptionView extends React.Component<Props, {}> {
  constructor(props: Props) {
    super(props);

    this.state = {};
  }

  public render() {
    return (
      <>
        {this.props.description.elements.map((element, idx) => {
          if (element.isText()) {
            return <TextElement model={element} key={idx} />;
          }

          if (element.isLink()) {
            return <LinkElement model={element} hasAccess={this.hasAccessToLink(element.targetId)} key={idx} />;
          }
        })}
      </>
    );
  }

  private hasAccessToLink(targetId: string): boolean {
    return AuthorizationManager.hasAccessToPerspective(targetId);
  }
}

export function TextElement(props: { model: CardDescriptionTextElement }) {
  return <span data-field="text">{props.model.text}</span>;
}

export function LinkElement(props: { model: CardDescriptionLinkElement; hasAccess: boolean }) {
  const disabledClass = props.hasAccess ? "" : "disabled";
  const onClickFunc = props.hasAccess
    ? (e: React.MouseEvent<HTMLElement>) => {
        e.stopPropagation();
        AppFormer.goTo(props.model.targetId);
      }
    : undefined;

  return (
    <a data-field="link" className={`kie-hero-card__link ${disabledClass}`} onClick={onClickFunc}>
      {props.model.text}
    </a>
  );
}
