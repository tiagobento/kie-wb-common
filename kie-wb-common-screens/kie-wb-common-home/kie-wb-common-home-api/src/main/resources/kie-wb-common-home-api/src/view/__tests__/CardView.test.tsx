import { AuthorizationManager } from "../../util";
import { Card, CardDescriptionBuilder } from "../../model";
import { shallow } from "enzyme";
import { CardView } from "../CardView";
import * as React from "react";
import * as AppFormer from "appformer-js";
import { CardDescriptionView } from "../CardDescriptionView";

describe("CardView", () => {
  describe("behaviour", () => {
    afterEach(() => {
      jest.resetAllMocks();
    });

    test("with access granted, should render consistently and set an onClick function", () => {
      jest.spyOn(AuthorizationManager, "hasAccessToPerspective").mockImplementation(() => true);

      const iconCssClasses = ["foo-clasz", "bar-clasz"];
      const title = "welcome!";
      const description = new CardDescriptionBuilder("hey!").build();
      const perspectiveId = "foo-perspective";
      const onMayClick = jest.fn(() => true);

      const model = new Card(iconCssClasses, title, description, perspectiveId, onMayClick);

      const component = shallow(<CardView model={model} />);

      // check root div properties
      expect(component.prop("data-field")).toEqual("card");
      expect(component.prop("className")).toEqual("kie-hero-card");
      expect(component.prop("role")).toEqual("button");
      expect(component.prop("id")).toEqual("home-action-welcome!");
      expect(component.prop("onClick")).toBeDefined();

      expect(component.children()).toHaveLength(2);

      const iconElement = component.childAt(0);
      expect(iconElement.children()).toHaveLength(0);
      expect(iconElement.prop("data-field")).toEqual("icon");
      expect(iconElement.prop("className")).toEqual(
        "kie-hero-card__icon kie-circle-icon kie-circle-icon--lg foo-clasz bar-clasz"
      );

      const textElement = component.childAt(1);
      expect(textElement.prop("className")).toEqual("kie-hero-card__text");
      expect(textElement.children()).toHaveLength(2);

      const titleElement = textElement.childAt(0);
      expect(titleElement.prop("data-field")).toEqual("heading");
      expect(titleElement.text()).toEqual("welcome!");

      const descriptionElement = textElement.childAt(1);
      expect(descriptionElement.type()).toEqual(CardDescriptionView);
      expect(descriptionElement.prop("description")).toEqual(description);
    });

    test("with access not granted, should render consistently and don't set an onClick function", () => {
      jest.spyOn(AuthorizationManager, "hasAccessToPerspective").mockImplementation(() => false);

      const iconCssClasses = ["foo-clasz", "bar-clasz"];
      const title = "welcome!";
      const description = new CardDescriptionBuilder("hey!").build();
      const perspectiveId = "foo-perspective";
      const onMayClick = jest.fn(() => true);

      const model = new Card(iconCssClasses, title, description, perspectiveId, onMayClick);

      const component = shallow(<CardView model={model} />);

      // check root div properties
      expect(component.prop("data-field")).toEqual("card");
      expect(component.prop("className")).toEqual("kie-hero-card");
      expect(component.prop("role")).toEqual("button");
      expect(component.prop("id")).toEqual("home-action-welcome!");
      expect(component.prop("onClick")).toBeUndefined();

      expect(component.children()).toHaveLength(2);

      const iconElement = component.childAt(0);
      expect(iconElement.children()).toHaveLength(0);
      expect(iconElement.prop("data-field")).toEqual("icon");
      expect(iconElement.prop("className")).toEqual(
        "kie-hero-card__icon kie-circle-icon kie-circle-icon--lg foo-clasz bar-clasz"
      );

      const textElement = component.childAt(1);
      expect(textElement.prop("className")).toEqual("kie-hero-card__text");
      expect(textElement.children()).toHaveLength(2);

      const titleElement = textElement.childAt(0);
      expect(titleElement.prop("data-field")).toEqual("heading");
      expect(titleElement.text()).toEqual("welcome!");

      const descriptionElement = textElement.childAt(1);
      expect(descriptionElement.type()).toEqual(CardDescriptionView);
      expect(descriptionElement.prop("description")).toEqual(description);
    });

    test("with access granted and onMayClick returning true, should call goTo to target perspective", () => {
      jest.spyOn(AuthorizationManager, "hasAccessToPerspective").mockImplementation(() => true);
      const goToSpy = jest.spyOn(AppFormer, "goTo");

      const iconCssClasses = ["foo-clasz", "bar-clasz"];
      const title = "welcome!";
      const description = new CardDescriptionBuilder("hey!").build();
      const perspectiveId = "foo-perspective";
      const onMayClick = jest.fn(() => true);

      const model = new Card(iconCssClasses, title, description, perspectiveId, onMayClick);

      const component = shallow(<CardView model={model} />);

      // check card onClick's behaviour
      component.simulate("click");
      expect(onMayClick).toBeCalledTimes(1);
      expect(goToSpy).toBeCalledTimes(1);
      expect(goToSpy).toHaveBeenCalledWith("foo-perspective");
    });

    test("with access granted and onMayClick returning false, should not call goTo to target perspective", () => {
      jest.spyOn(AuthorizationManager, "hasAccessToPerspective").mockImplementation(() => true);
      const goToSpy = jest.spyOn(AppFormer, "goTo");

      const iconCssClasses = ["foo-clasz", "bar-clasz"];
      const title = "welcome!";
      const description = new CardDescriptionBuilder("hey!").build();
      const perspectiveId = "foo-perspective";
      const onMayClick = jest.fn(() => false);

      const model = new Card(iconCssClasses, title, description, perspectiveId, onMayClick);

      const component = shallow(<CardView model={model} />);

      // check card onClick's behaviour
      component.simulate("click");
      expect(onMayClick).toBeCalledTimes(1);
      expect(goToSpy).not.toBeCalled();
    });

    test("with access granted and onMayClick undefined, should not call goTo to target perspective", () => {
      jest.spyOn(AuthorizationManager, "hasAccessToPerspective").mockImplementation(() => true);
      const goToSpy = jest.spyOn(AppFormer, "goTo");

      const iconCssClasses = ["foo-clasz", "bar-clasz"];
      const title = "welcome!";
      const description = new CardDescriptionBuilder("hey!").build();
      const perspectiveId = "foo-perspective";

      const model = new Card(iconCssClasses, title, description, perspectiveId);

      const component = shallow(<CardView model={model} />);

      // check card onClick's behaviour
      component.simulate("click");
      expect(goToSpy).toHaveBeenCalledTimes(1);
    });

    test("with access granted and onMayClick returning false, should not call goTo to target perspective", () => {
      jest.spyOn(AuthorizationManager, "hasAccessToPerspective").mockImplementation(() => true);
      const goToSpy = jest.spyOn(AppFormer, "goTo");

      const iconCssClasses = ["foo-clasz", "bar-clasz"];
      const title = "welcome!";
      const description = new CardDescriptionBuilder("hey!").build();
      const perspectiveId = "foo-perspective";
      const onMayClick = jest.fn(() => false);

      const model = new Card(iconCssClasses, title, description, perspectiveId, onMayClick);

      const component = shallow(<CardView model={model} />);

      // check card onClick's behaviour
      component.simulate("click");
      expect(onMayClick).toBeCalledTimes(1);
      expect(goToSpy).not.toBeCalled();
    });

    test("with access not granted, should call goTo to target perspective", () => {
      jest.spyOn(AuthorizationManager, "hasAccessToPerspective").mockImplementation(() => false);
      const goToSpy = jest.spyOn(AppFormer, "goTo");

      const iconCssClasses = ["foo-clasz", "bar-clasz"];
      const title = "welcome!";
      const description = new CardDescriptionBuilder("hey!").build();
      const perspectiveId = "foo-perspective";
      const onMayClick = jest.fn(() => true);

      const model = new Card(iconCssClasses, title, description, perspectiveId, onMayClick);

      const component = shallow(<CardView model={model} />);

      // check card onClick's behaviour
      component.simulate("click");
      expect(onMayClick).not.toBeCalled();
      expect(goToSpy).not.toBeCalled();
    });
  });

  describe("snapshot", () => {
    afterEach(() => {
      jest.resetAllMocks();
    });

    test("with access granted, should render consistently", () => {
      jest.spyOn(AuthorizationManager, "hasAccessToPerspective").mockImplementation(() => true);

      const iconCssClasses = ["foo-clasz", "bar-clasz"];
      const title = "welcome!";
      const description = new CardDescriptionBuilder("hey!").build();
      const perspectiveId = "foo-perspective";
      const onMayClick = jest.fn(() => true);

      const model = new Card(iconCssClasses, title, description, perspectiveId, onMayClick);

      const component = shallow(<CardView model={model} />);

      expect(component).toMatchSnapshot();
    });

    test("with access not granted, should render consistently", () => {
      jest.spyOn(AuthorizationManager, "hasAccessToPerspective").mockImplementation(() => false);

      const iconCssClasses = ["foo-clasz", "bar-clasz"];
      const title = "welcome!";
      const description = new CardDescriptionBuilder("hey!").build();
      const perspectiveId = "foo-perspective";
      const onMayClick = jest.fn(() => true);

      const model = new Card(iconCssClasses, title, description, perspectiveId, onMayClick);

      const component = shallow(<CardView model={model} />);

      expect(component).toMatchSnapshot();
    });
  });
});
