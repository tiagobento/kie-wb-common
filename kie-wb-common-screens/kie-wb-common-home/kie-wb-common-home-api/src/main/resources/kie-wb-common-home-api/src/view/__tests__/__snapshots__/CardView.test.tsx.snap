// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardView snapshot with access granted, should render consistently 1`] = `
<div
  className="kie-hero-card"
  data-field="card"
  id="home-action-welcome!"
  onClick={[Function]}
  role="button"
>
  <div
    className="kie-hero-card__icon kie-circle-icon kie-circle-icon--lg foo-clasz bar-clasz"
    data-field="icon"
  />
  <div
    className="kie-hero-card__text"
  >
    <h2
      data-field="heading"
    >
      welcome!
    </h2>
    <CardDescriptionView
      description={
        CardDescription {
          "elements": Array [
            CardDescriptionTextElement {
              "text": "hey!",
            },
          ],
        }
      }
    />
  </div>
</div>
`;

exports[`CardView snapshot with access not granted, should render consistently 1`] = `
<div
  className="kie-hero-card"
  data-field="card"
  id="home-action-welcome!"
  role="button"
>
  <div
    className="kie-hero-card__icon kie-circle-icon kie-circle-icon--lg foo-clasz bar-clasz"
    data-field="icon"
  />
  <div
    className="kie-hero-card__text"
  >
    <h2
      data-field="heading"
    >
      welcome!
    </h2>
    <CardDescriptionView
      description={
        CardDescription {
          "elements": Array [
            CardDescriptionTextElement {
              "text": "hey!",
            },
          ],
        }
      }
    />
  </div>
</div>
`;
